lib/valkyrie/persistence/postgres/query_service.rb

```diff
@@ -25,7 +25,7 @@ def find_all_of_model(model:)
     # (see Valkyrie::Persistence::Memory::QueryService#find_by)
     def find_by(id:)
       validate_id(id)
-      resource_factory.to_resource(object: orm_class.find(id))
+      resource_factory.to_resource(object: orm_class.find(id.to_s))
```

Comment:
Why do we need the casting here?

Reply:
I had problems with it whenever it tries to interpolate the query parameters. When the database's field type is string, it wasn't calling to_s on the Valkyrie::ID. Had to explicitly call it.